optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/reflect/generic/TestdataGenericEntity.java

```diff
@@ -59,10 +60,19 @@ public TestdataGenericEntity(String code, TestdataGenericValue value) {
         return value;
     }
 
+    @PlanningVariable(valueRangeProviderRefs = "subTypeValueRange")
+    public TestdataGenericValue<T> getSubTypeValue() {
+        return subTypeValue;
+    }
+
     public void setValue(TestdataGenericValue<T> value) {
         this.value = value;
     }
 
+    public void setSubTypeValue(TestdataGenericValue<T> subTypeValue) {
```

Comment:
Are there any test methods using this?

Reply:
Maybe I need some help in understanding the ask here. 

As per my understanding, the use of Java Generic in solution is tested in the following test method org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptorTest#generic.

When setSubTypeValue setter is removed this test case fails with the following error.

`
java.lang.IllegalStateException: The class (class org.optaplanner.core.impl.testdata.domain.reflect.generic.TestdataGenericEntity) has a PlanningVariable annotated getter method (public org.optaplanner.core.impl.testdata.domain.reflect.generic.TestdataGenericValue org.optaplanner.core.impl.testdata.domain.reflect.generic.TestdataGenericEntity.getSubTypeValue()), but lacks a setter for that property (subTypeValue).
`

Thus what I understand here is that this setter is not directly used in the test cases but fails because of existing getter and setter validation for Planning Variables at Planning Entity level.

To make the asserts better in this generic test method, I have tried to add additional asserts in this test method to validate Value Range Providers at the solution and Planning Variable at entity level. Let me know if that helps.

Reply:
Let me know if I can help to resolve this conversation? So that we can merge the PR. Will be good to see this fix in next release.

Reply:
My apologies, I didn't notice that the SolutionDescriptorTest did use this new class. https://github.com/kiegroup/optaplanner/pull/654/files#diff-5b1d1d2d1eaa460524ed6b58b40365b9R166
Yes, this PR is perfect. Merging now. Sorry for the delay. The bad news it just missed 7.32.0.Final... (it got branches on monday, release is baking now) but it will be in 7.33.0.Final (normally 3-4 weeks later but that's not a promise).